src/libotutil/ot-checksum-instream.c

```diff
@@ -68,6 +84,12 @@ ot_checksum_instream_class_init (OtChecksumInstreamClass *klass)
 
   object_class->finalize = ot_checksum_instream_finalize;
   stream_class->read_fn = ot_checksum_instream_read;
+
+# if defined(HAVE_GCRYPT)
+  g_assert (gcry_check_version (GCRYPT_VERSION));
+  gcry_control (GCRYCTL_DISABLE_SECMEM, 0);
+  gcry_control (GCRYCTL_INITIALIZATION_FINISHED, 0);
```

Comment:
Not your fault - this bit is recommended by the docs, but I have to say this is ugly.  Do we really need to do this dance to just use the checksums API?

Also, doesn't it potentially conflict with using libgpgme in process to e.g. sign things?  Hopefully calling `GCRYCTL_DISABLE_SECMEM` doesn't like deliberately do a `open('/tmp/myprivategpgkey').write(...).chmod(0644)` or whatever but still...

Reply:
It sounds like you are planning to use gnutls here and not libgcrypt; if that's the case can we just omit that for now?

Reply:
Yes, unfortunately it is necessary, otherwise some of the hash stuff fails on "make check" with GPG_ERR_NOT_OPERATIONAL...

I can remove the libgcrypt parts if it is seen best alternative. Initially gnutls seems to be most convenient way to do the certificate and signature functionality. And it is sort of supported by the fact that libsoup uses it too (when/if used with ostree).

I call DISABLE_SECMEM to avoid issues when doing containerized builds or builds on restricted environments that don't allow mlock() and similar. My thinking was that build servers are not high risk environments for key leakage and at client side public keys are not sensitive.